[docs] Improve the Quick Start dev setup section (#36646)
cilium-team/cilium: Documentation/contributing/development/dev_setup.rst

Quick Start
-----------

> The below commands install Cilium within a `Kind`_-based Kubernetes cluster.

Reviewer: s/below/following, clear plain language
```suggestion
The following commands install Cilium in a `Kind`_-based Kubernetes cluster.
```

> Invoke them within the root of the Cilium repository. Their output will inform
> you of any missing dependencies (note that "Cilium not being found" refers to
> the Cilium CLI). The make targets are described in later sections.

Reviewer: "Execute" is standard, but you can also use "run", which I slightly prefer.
Specific is better than general.
Use clear, plain language.
Use present tense.
Wrap commands in code font.
What's the actual message that users will see for "Cilium not being found"? Better to quote that and separate the "Note" sentence into a `.. note::` callout.
Which later sections? List them, preferably with linked text.
```suggestion
Run them in the root directory of the Cilium repository. The command output informs
you of any missing dependencies. (Note that "Cilium not being found" refers to
the Cilium CLI.) The `make` targets are described in later sections.
```

.. _Kind: https://kind.sigs.k8s.io/

On Linux:

.. code-block:: shell-session

   make kind

>    make kind-image-fast

Reviewer 2: I notice you've changed the order - I think if you `install` first and then build the images, you'll be running the images you've built, whereas if you build first then install you'll be running images from latest `main` CI, until you build again - is this an intentional change?

Author: I addressed this in the description. I chose to swap the order because the `make kind` output says to use that order and also because the generic (non-Linux specific) commands also use this order. It would be good to get a second opinion however and I thank you for your concern.

>    make kind-install-cilium-fast

Reviewer: Nice, for this and all similar changes. ✨
